shared/remediations/bash/templates/remediation_functions

```diff
@@ -701,3 +701,87 @@ do
 done
 
 }
+
+# Function to replace configuration setting in config file or add the configuration setting if
+# it does not exist.
+#
+# Expects four arguments:
+#
+# config_file:		Configuration file that will be modified
+# key:			Configuration option to change
+# value:		Value of the configuration option to change
+# cce:			The CCE identifier or 'CCENUM' if no CCE identifier exists
+#
+# Optional arugments:
+#
+# format:		Optional argument to specify the format of how key/value should be
+# 			modified/appended in the configuration file. The default is key = value.
+#
+# Example Call(s):
+#
+#     With default format of 'key = value':
+#     replace_or_append '/etc/sysctl.conf' '^kernel.randomize_va_space' '2' 'CCENUM'
+#
+#     With custom key/value format:
+#     replace_or_append '/etc/sysconfig/selinux' '^SELINUX=' 'disabled' 'CCENUM' '%s=%s'
+#
+#     With a variable:
+#     replace_or_append '/etc/sysconfig/selinux' '^SELINUX=' $var_selinux_state 'CCENUM' '%s=%s'
+#
+function replace_or_append {
+  local config_file=$1
+  local key=$2
+  local value=$3
+  local cce=$4
+  local format=$5
+
+  # Check sanity of the input
+  if [ $# -lt "3" ]
+  then
+        echo "Usage: replace_or_append 'config_file_location' 'key_to_search' 'new_value'"
+        echo
+        echo "If symlinks need to be taken into account, add yes/no to the last argument"
+        echo "to allow to 'follow_symlinks'."
+        echo "Aborting."
+        exit 1
+  fi
+
+  # Test if the config_file is a symbolic link. If so, use --follow-symlinks with sed.
+  # Otherwise, regular sed command will do.
+  if test -L $config_file; then
+    sed_command="sed --follow-symlinks"
+  else
+    sed_command="sed"
+  fi
+
+  # Test that the cce arg is not empty or does not equal CCENUM.
+  # If CCENUM exists, it means that there is no CCE assigned.
+  if ! [ "x$cce" = x ] && [ "$cce" != 'CCENUM' ]; then
+    cce="CCE-${cce}"
+  else
+    cce="CCE"
+  fi
+
+  # Strip any search characters in the key arg so that the key can be replaced without
+  # adding any search characters to the config file.
+  stripped_key=${key//[!a-z]/}
+
+  # If there is no print format specified in the last arg, use the default format.
+  if ! [ "x$format" = x ] ; then
+    printf -v formatted_output $format $stripped_key $value
+  else
+    formatted_output="$stripped_key = $value"
+  fi
+
+  # If the key exists, change it. Otherwise, add it to the config_file.
+  if grep -qi $key $config_file ; then
+    $sed_command "s/$key.*/$formatted_output/g" $config_file
+  else
+#    echo -ne "\n# Per $cce: Set $formatted_output in $config_file" >> $config_file
```

Comment:
Last (but not least): This one looks like temporary commented out code, which has been forgotten to be un-commented. And the two code statements at lines #782 and #783 were expected to be deleted IMHO in the final version. Because without doing these changes you again end up just printing the changes (the modified content of the final file) to the stdout, rather than to actually inserting the changes into the file.